src/main/scala/higherkindness/rules_scala/workers/common/CommonArguments.scala

```diff
@@ -49,14 +61,18 @@ object Analysis {
 }
 
 object CommonArguments {
+  private val scala2SemanticDbTargetRootRegex = """-P:semanticdb:targetroot:(.*)""".r
+  private val scala3SemanticDbTargetRootRegex = """-semanticdb-target:(.*)""".r
+
   private def adjustCompilerOptions(workDir: Path, options: List[String]) = {
-    def adjustStringPath(path: String) = SandboxUtil.getSandboxPath(workDir, Paths.get(path)).toString
+    def adjustStringPath(path: String) =
+      SandboxUtil.getSandboxPath(workDir, Paths.get(path)).toString
 
     options.flatMap {
-      case s"-P:semanticdb:targetroot:$path" =>
+      case scala2SemanticDbTargetRootRegex(path) =>
         List(s"-P:semanticdb:sourceroot:${workDir.toString}", s"-P:semanticdb:targetroot:${adjustStringPath(path)}")
 
-      case s"-semanticdb-target:$path" =>
+      case scala3SemanticDbTargetRootRegex(path) =>
         List(s"-semanticdb-target:${adjustStringPath(path)}", s"-sourceroot:${workDir.toString}")
```

Comment:
I forgot to ask this when we added the semanticdb functionality, but does it matter if the `sourceroot` ceases to exist after the semanticdb call? Because there's no guarantee the sandbox directory will exist long term.

Reply:
I don't think so. SemanticDB information is separate from sources and from what I understand, the SemanticDB files don't include the source root path. The sandbox directory only needs to live as long as the compiler.